File: BlockchainJava/BlockchainFundamentals/src/com/blockchain/Transaction.java
package com.blockchain;


/**
 * 
 * @author anubhp
 * 
 * sender: address of the sender.
 * receiver: address of the receiver.
 * amount: amount/number of money/coins transferred.
 *
 */
public class Transaction {
	private String sender;
	private String receiver;
	private double amount;
	public Transaction(String sender, String receiver, double amount) {
		super();
		this.sender = sender;
		this.receiver = receiver;
		this.amount = amount;
	}
	public String getSender() {
		return sender;
	}
	public void setSender(String sender) {
		this.sender = sender;
	}
	public String getReceiver() {
		return receiver;
	}
	public void setReceiver(String receiver) {
		this.receiver = receiver;
	}
	public double getAmount() {
		return amount;
	}
	public void setAmount(double amount) {
		this.amount = amount;
	}
	@Override
	public String toString() {
		return "Transaction [sender=" + sender + ", receiver=" + receiver + ", amount=" + amount + "]";
	}
	
}

File: BlockchainCpp/Transaction.cpp
#include "Transaction.h"
#include "utils.h"

using namespace std;
using namespace utils;

Transaction::Transaction(string sender, string receiver, double amount) {
    this->sender = sender;
    this->receiver = receiver;
    this->amount = amount;
}

string Transaction::getSender() {
	return this->sender;
}
void Transaction::setSender(string sender) {
    this->sender = sender;
}
string Transaction::getReceiver() {
    return this->receiver;
}
void Transaction::setReceiver(string receiver) {
    this->receiver = receiver;
}
double Transaction::getAmount() {
    return this->amount;
}
void Transaction::setAmount(double amount) {
    this->amount = amount;
}

string Transaction::toString(){
    return "Transaction [sender=" + sender + ", receiver=" + receiver + ", amount=" + to_String(amount) + "]";
}